File: backend/internal/models/season.go
package models

import "time"

type Season struct {
	DbModel
	Name          string         `json:"name" validate:"required"`
	Start         time.Time      `json:"start" validate:"required"`
	End           time.Time      `json:"end" validate:"required"`
	Registrations []Registration `json:"registrations"`
	Schedule      []Game         `json:"schedule"`
	Leagues       []League       `json:"leagues"`
}

File: frontend/models/Season.ts
import type { TimeString } from './types/timeString'
import type { Game } from './Game'
import type { Roster } from './Roster'

export interface Season {
  // dbModel: DbModel
  name: string
  start: TimeString
  end: TimeString
  schedule: Array<Game>
  // registrations: Array<Registration>
  rosters: Array<Roster>
}
